🛠 Tooling: End-to-end migration test lint takes a long time on coverage-migrate/ files

### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

I noticed while tinkering with #1048 that `pnpm run test:migrate` spends a lot of time locally in _"Cleaning up files"_. That log wraps around two commands:

https://github.com/JoshuaKGoldberg/create-typescript-app/blob/c06c62c283b44df22a41dbecdf3dfb515316a399/src/migrate/migrateWithOptions.ts#L63-L66

I tried running `pnpm lint` locally... and it reported quite a few issues:

```plaintext
✖ 457725 problems (457725 errors, 0 warnings)
  457696 errors and 0 warnings potentially fixable with the `--fix` option.
```

Wow. It features lovely lines like this:

```plaintext
/Users/josh/repos/create-typescript-app/coverage-migrate/tmp/coverage-48900-1703919745073-0.json
  1:74       error  Expected object keys to be in ascending order. 'functions' should be before 'url'                                                                                                                                                                                                                                                                                                                                                                                                        jsonc/sort-keys
```

...along with roughly 73,000 more errors.

This explains the slowness! Because the migration script runs `pnpm lint --fix` when it's done, and the migration script puts coverage reports in `coverage-migrate`... there's now a whole bunch of linting being done.

### Additional Info

Splitting this performance improvement task out of #860.

I'm not sure exactly what the right fix is here...

--- a/.github/workflows/test-initialize.yml
+++ b/.github/workflows/test-initialize.yml
@@ -6,6 +6,10 @@ jobs:
       - uses: ./.github/actions/prepare
       - run: pnpm run build
       - run: pnpm run test:initialize
+      # The template's .eslintignore only ignores coverage, not coverage-*
+      # https://github.com/JoshuaKGoldberg/create-typescript-app/issues/1131
+      - if: always()
+        run: mv coverage coverage-initialize
       - if: always()
         name: Codecov
         uses: codecov/codecov-action@v3

--- a/.github/workflows/test-migrate.yml
+++ b/.github/workflows/test-migrate.yml
@@ -6,6 +6,10 @@ jobs:
       - uses: ./.github/actions/prepare
       - run: pnpm run build
       - run: pnpm run test:migrate
+      # The template's .eslintignore only ignores coverage, not coverage-*
+      # https://github.com/JoshuaKGoldberg/create-typescript-app/issues/1131
+      - if: always()
+        run: mv coverage coverage-migrate
       - if: always()
         name: Codecov
         uses: codecov/codecov-action@v3

--- a/script/initialize-test-e2e.js
+++ b/script/initialize-test-e2e.js
@@ -49,4 +49,4 @@ await $`pnpm i`;
 await $`pnpm run build`;
 await $({
 	stdio: "inherit",
-})`c8 -o ./coverage-initialize -r html -r lcov --src src node ./bin/index.js --base everything --description ${description} --mode initialize --owner ${owner} --title ${title} --repository ${repository} --skip-all-contributors-api --skip-github-api --skip-removal --skip-restore`;
+})`c8 -o ./coverage -r html -r lcov --src src node ./bin/index.js --base everything --description ${description} --mode initialize --owner ${owner} --title ${title} --repository ${repository} --skip-all-contributors-api --skip-github-api --skip-removal --skip-restore`;

--- a/script/migrate-test-e2e.js
+++ b/script/migrate-test-e2e.js
@@ -11,7 +11,7 @@ const title = "Create TypeScript App";
 
 await $({
 	stdio: "inherit",
-})`c8 -o ./coverage-migrate -r html -r lcov --src src node ./bin/index.js --base everything --mode migrate --description ${description} --email-github ${emailGithub} --email-npm ${emailNpm} --owner ${owner} --title ${title} --repository ${repository} --skip-all-contributors-api --skip-github-api --skip-install`;
+})`c8 -o ./coverage -r html -r lcov --src src node ./bin/index.js --base everything --mode migrate --description ${description} --email-github ${emailGithub} --email-npm ${emailNpm} --owner ${owner} --title ${title} --repository ${repository} --skip-all-contributors-api --skip-github-api --skip-install`;
 
 const { stdout: gitStatus } = await $`git status`;
 console.log(`Stdout from running \`git status\`:\n${gitStatus}`);

